llama-index-core/llama_index/core/memory/chat_memory_buffer.py

```diff
@@ -109,20 +109,26 @@ def get(self, initial_token_count: int = 0, **kwargs: Any) -> List[ChatMessage]:
             raise ValueError("Initial token count exceeds token limit")
 
         message_count = len(chat_history)
-        token_count = (
-            self._token_count_for_message_count(message_count) + initial_token_count
-        )
+
+        cur_messages = chat_history[-message_count:]
```

Comment:
i think this implementation works. a more optimal method (as suggested by #10493) is to implement some data structure on top of memory that allows for faster seeking as opposed to calling `get_all` at all

Reply:
I think to have a faster "seek", I think we'd have to refactor how chat history is stored (maybe key-value pairs of messages and token counts, stored as a linked list or similar?)

For now, this version should be way faster (`get_all()` called only once now)